pkg/apis/resource/types.go

```diff
@@ -0,0 +1,449 @@
+	SelectedNode string
+
+	// When allocation is delayed, and the scheduler needs to decide on
+	// which node a Pod should run, it will ask the driver(s) on which
+	// nodes the resource might be made available. To trigger that check,
+	// the scheduler provides the names of nodes which might be suitable
+	// for the Pod. Will be updated periodically until all resources are
+	// allocated.
+	//
+	// The ResourceClass.SuiteableNodes node selector can be used to filter
+	// out nodes based on labels. This prevents adding nodes here that the
+	// driver then would need to reject through UnsuitableNodes.
+	//
+	// The size of this field is limited to 128. This is large enough for
+	// many clusters. Larger clusters may need more attempts to find a node
+	// that suits all pending resources.
+	PotentialNodes []string
+}
+
+// PodSchedulingStatus is where resource drivers provide information about
+// where the could allocate a resource and whether allocation failed.
+type PodSchedulingStatus struct {
+	// Each resource driver is responsible for providing information about
+	// those resources in the Pod that the driver manages. It can skip
+	// adding that information when it already allocated the resource.
+	//
+	// A driver must add entries here for all its pending claims, even if
+	// the ResourceSchedulingStatus.UnsuitabeNodes field is empty, because
+	// the scheduler may decide to wait with selecting a node until it has
+	// information from all drivers.
+	ResourceClaims []ResourceClaimSchedulingStatus
+
+	// If there ever is a need to support other kinds of resources
+	// than ResourceClaim, then new fields could get added here
+	// for those other resources.
+}
+
+// ResourceClaimSchedulingStatus contains information about one particular
+// claim while scheduling a pod.
+type ResourceClaimSchedulingStatus struct {
+	// PodClaimName matches the PodClaim.Name field.
+	PodClaimName string
+
+	// UnsuitableNodes lists nodes that the claim cannot be allocated for.
+	// Nodes listed here will be ignored by the scheduler when selecting a
+	// node for a Pod. All other nodes are potential candidates, either
+	// because no information is available yet or because allocation might
+	// succeed.
+	//
+	// A change of the PodSchedulingSpec.PotentialNodes field and/or a failed
+	// allocation attempt trigger an update of this field: the driver
+	// then checks all nodes listed in PotentialNodes and UnsuitableNodes
+	// and updates UnsuitableNodes.
+	//
+	// It must include the prior UnsuitableNodes in this check because the
+	// scheduler will not list those again in PotentialNodes but they might
+	// still be unsuitable.
+	//
+	// This can change, so the driver also must refresh this information
+	// periodically and/or after changing resource allocation for some
+	// other ResourceClaim until a node gets selected by the scheduler.
+	//
+	// The size of this field is limited to 128, the same as for
+	// PodSchedulingSpec.PotentialNodes.
```

Comment:
This should be something like:
```golang
	// For `WaitForFirstConsumer` allocations: a subset of the candidate notes where a
	// scheduler should not attempt to place this Pod.
	//
	// For `Immediate` allocations, ???
	//
	// When dynamic resource allocation fails, the failed allocation typically triggers an update
	// to add the name of the failed node into this list.
	// The API server may limit the number of entries in this list (returning a ??? response status
	// if a client requests adding an entry beyond the API server's capacity for entries here).
	// If adding an unsuitable node name fails, the client
	// that attempted the addition should silently ignore that error.
	SelectedNode string
```

but the existing comment wasn't clear enough about what this should be for the immediate allocation case for me to make a suggestion.

Don't mention drivers doing periodic refreshes unless we know that a **watch** verb could not ever work.

Reply:
See above.